src/components/Products/GridItem/GridItem.js

```diff
@@ -15,110 +15,117 @@ import { getIsFavoriteStatusFromProducts } from 'store/products/selectors';
-          <div className="gridItem__description">
-            <p className="gridItem__text">{name}</p>
-            <p className="gridItem__text">${price / 100}</p>
-          </div>
-          <div className="gridItem__first-icon">
-            <ButtonRound
-              icon={iconFav}
-              onClick={favoriteToggleHandler}
-              info={favoriteInfoText}
-              className={favoriteClasses}
-            />
-          </div>
-          <div className="gridItem__show-icon">
-            <ButtonRound
-              icon={iconDetail}
-              onClick={showDetailModalHandler}
-              info="show details"
-              className="button-round"
-            />
-          </div>
-          <div className="gridItem__show-button">
-            <ButtonLong
-              title="Add to cart"
-              onClick={addToCartHandler}
-              icon={iconBtn}
-            />
-          </div>
-        </div>
-      </li>
-      {isDetailModalVisible && (
-        <Modal
-          element={
-            <ShowProductDetail
-              product={{ imageUrl, name, price, id, description, inStock }}
-            />
-          }
-          onClick={closeDetailsModalHandler}
-        />
-      )}
-    </>
-  );
+    return (
+        <>
+            <li className="gridItem-container" onClick={redirectToDetailPage}>
+                <div className="gridItem">
+                    <div className="gridItem__img-wrap">
+                        <img src={imageUrl} />
+                    </div>
+                    <div className="gridItem__description">
+                        <p className="gridItem__text">{name}</p>
+                        <p className="gridItem__text">${price / 100}</p>
+                    </div>
+                    <div className="gridItem__first-icon">
+                        <ButtonRound
+                            icon={iconFav}
+                            onClick={favoriteToggleHandler}
+                            info={favoriteInfoText}
+                            className={favoriteClasses}
+                        />
+                    </div>
+                    <div className="gridItem__show-icon">
+                        <ButtonRound
+                            icon={iconDetail}
+                            onClick={showDetailModalHandler}
+                            info="show details"
+                            className="button-round"
+                        />
+                    </div>
+                    <div className="gridItem__show-button">
+                        <ButtonLong
+                            title="Add to cart"
+                            onClick={addToCartHandler}
+                            icon={iconBtn}
+                        />
+                    </div>
+                </div>
+            </li>
+            {isDetailModalVisible && (
+                <Modal
+                    element={
+                        <ShowProductDetail
```

Comment:
Product details modal broken on mobile, it's impossible to scroll down.

![Screenshot 2023-03-15 at 21 27 02](https://user-images.githubusercontent.com/2543423/225434302-a99e05ca-df2a-4920-8497-5c8753d1792e.png)